plugins/uefi-capsule/fu-uefi-device.c

```diff
@@ -537,6 +537,70 @@ fu_uefi_device_cleanup (FuDevice *device,
 	return TRUE;
 }
 
+static gboolean
+fu_check_capsuleupdate_ondisk (FuDevice *device, GError **error)
+{
+	g_autofree guint64 *os_indications_supported = NULL;
+	guint32 attr;
+	gsize os_ind_size = 0;
+
+	if (!fu_device_get_metadata_boolean (device, "DisableCapsuleUpdateOnDisk")) {
+		g_set_error_literal (error,
+			     FWUPD_ERROR,
+                             FWUPD_ERROR_NOT_SUPPORTED,
+                             "Capsule update on-disk is disabled");
+		return FALSE;
+	}
+
+	/* check for capsuleupdate ondisk */
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL,
+				 "OsIndicationsSupported",
+				 (guint8 **) &os_indications_supported,
+				 &os_ind_size, &attr, error)) {
+		g_prefix_error (error, "Failed to read OsIndicationsSupported: ");
+		return FALSE;
+	}
+
+	if (!(*os_indications_supported &
+	      EFI_OS_INDICATIONS_FILE_CAPSULE_DELIVERY_SUPPORTED)) {
+		g_set_error_literal (error,
+			     FWUPD_ERROR,
+                             FWUPD_ERROR_NOT_SUPPORTED,
+                             "Firmware doesn't support capsule update on-disk");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_capsule_update_on_disk (FuDevice *device, GError **error)
+{
+	g_autofree guint64 *os_indications = NULL;
+	guint32 attr;
+	gsize os_ind_size = 0;
+
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "OsIndications",
+			 (guint8 **) &os_indications,
+			 &os_ind_size, &attr, error)) {
```

Comment:
you need to use a local error here or return false in this if loop result

Reply:
Since this isn't an error that should prevent us from running, I'll set a local error and ignore it